src/container/srv_target.c

```diff
@@ -56,20 +54,21 @@ agg_rate_ctl(void *arg)
 	struct ds_pool		*pool = cont->sc_pool->spc_pool;
 	struct sched_request	*req = param->ap_req;
 
-	if (dss_ult_exiting(req))
+	if (dss_ult_exiting(req) || pool->sp_reclaim == DAOS_RECLAIM_DISABLED)
 		return true;
 
-	switch (pool->sp_reclaim) {
-	case DAOS_RECLAIM_DISABLED:
-		return true;
-	default:
-		if (dss_xstream_is_busy() &&
-		    sched_req_space_check(req) == SCHED_SPACE_PRESS_NONE)
-			sched_req_sleep(req, DAOS_AGG_LAZY_RATE);
+	/* System is busy and there is no space pressure */
+	if (dss_xstream_is_busy() && sched_req_space_check(req) == SCHED_SPACE_PRESS_NONE) {
+		/* Abort aggregation if reclaim mode is lazy, otherwise, take an extra sleep */
+		if (pool->sp_reclaim == DAOS_RECLAIM_LAZY)
+			return true;
```

Comment:
Hmm, then hae will not be updated if xstream become busy during aggregation. I thought keep sleeping here will not impact the performance? no?

Reply:
Right, HAE won't be updated since we aborted aggregation. I agree with you that sleep won't impact performance in theory, so we didn't have such change in master.

I think this is just a workaround for 2.0 before we solve all potential aggregation issues.

Reply:
> Right, HAE won't be updated since we aborted aggregation.

If the system turns to busy with certain interval, then hae will unlikely proceed.  will this add further burden later?

Sorry, a bit confused here, since sleep will not impact the performance, then why abort here, instead of sleep and update hae?